File: onboarding-java/src/main/java/com/hr/onboardingjava/auth/service/UserService.java
package com.hr.onboardingjava.auth.service;

import com.hr.onboardingjava.auth.dto.LoginRequest;
import com.hr.onboardingjava.auth.dto.LoginResponse;
import com.hr.onboardingjava.auth.dto.SignUpRequest;
import com.hr.onboardingjava.auth.dto.SignUpResponse;
import com.hr.onboardingjava.auth.entity.User;
import com.hr.onboardingjava.auth.repository.UserRepository;
import com.hr.onboardingjava.security.jwt.JwtPlugin;
import lombok.RequiredArgsConstructor;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@RequiredArgsConstructor
@Service
public class UserService {

    private final UserRepository userRepository;
    private final PasswordEncoder passwordEncoder;
    private final JwtPlugin jwtPlugin;

    public LoginResponse login(LoginRequest loginRequest) {
        User user = userRepository.findByUsername(loginRequest.getUsername());
        if (user != null && passwordEncoder.matches(loginRequest.getPassword(), user.getPassword())) {
            String token = jwtPlugin.generateAccessToken(user.getId().toString(),user.getUsername(),"USER");
            return LoginResponse.builder().token(token).build();
        }
        throw new RuntimeException("Invalid credentials");
    }

    @Transactional
    public SignUpResponse signUp(SignUpRequest signUpRequest) {
        User user = userRepository.findByUsername(signUpRequest.getUsername());
        if (user != null) {
            throw new RuntimeException("Username already exists");
        }

        User newUser = new User();
        newUser.setUsername(signUpRequest.getUsername());
        newUser.setPassword(passwordEncoder.encode(signUpRequest.getPassword()));
        newUser.setNickname(signUpRequest.getNickname());

        User savedUser = userRepository.save(newUser);
        return SignUpResponse.from(savedUser);
    }
}

File: onboarding-kt/src/main/kotlin/com/hr/onboardingkt/auth/service/UserService.kt
package com.hr.onboardingkt.auth.service

import com.hr.onboardingkt.auth.dto.LoginRequest
import com.hr.onboardingkt.auth.dto.LoginResponse
import com.hr.onboardingkt.auth.dto.SignUpRequest
import com.hr.onboardingkt.auth.dto.SignUpResponse
import com.hr.onboardingkt.auth.entity.Users
import com.hr.onboardingkt.auth.repository.UserRepository
import com.hr.onboardingkt.security.jwt.JwtPlugin
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class UserService (
    private val userRepository: UserRepository,
    private val jwtPlugin: JwtPlugin,
    private val passwordEncoder: PasswordEncoder
){
    fun login(
        loginRequest: LoginRequest
    ): LoginResponse {
        val user =userRepository.findByUsername(loginRequest.username)
        if(user != null && passwordEncoder.matches(loginRequest.password, user.password)) {
            val token = jwtPlugin.generateAccessToken(user.id.toString(), user.username,"USER")
            return LoginResponse(token)
        }
        throw RuntimeException("사용자 이름 또는 비밀번호가 잘못되었습니다.")
    }

    @Transactional
    fun signUp(
        signUpRequest: SignUpRequest
    ): SignUpResponse {
        userRepository.findByUsername(signUpRequest.username)?.let {
            throw RuntimeException("사용자 이름 '${signUpRequest.username}'은(는) 이미 존재합니다.")
        }

        val user = Users(
            username = signUpRequest.username,
            password = passwordEncoder.encode(signUpRequest.password),
            nickname = signUpRequest.nickname
        )

        val savedUser = userRepository.save(user)
        return SignUpResponse.from(savedUser)
    }
}